# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/xpconnect/src/XPCJSContext.cpp
# Commit: 9d31a829691c
# Full Hash: 9d31a829691ccf798375446e7ef04109e2662175
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2023-06-07 03:30:33
# Regressor Bug: 1835876
# File Overlap Count: 1
# Description:
#   Bug 1835876 part 2 - Disable code write protection in content processes. r=nbp
#   
#   We've worked with the security and performance teams to re-evaluate the W^X policy
#   we have in place to mark JIT code memory pages as either writable or executable
#   (but not both).
# ==============================================================================

diff -r 1ce5ad17da6b -r 9d31a829691c js/xpconnect/src/XPCJSContext.cpp
--- a/js/xpconnect/src/XPCJSContext.cpp	Tue Jun 06 12:16:46 2023 +0000
+++ b/js/xpconnect/src/XPCJSContext.cpp	Tue Jun 06 12:16:46 2023 +0000
@@ -969,6 +969,12 @@
           javascript_options_spectre_jit_to_cxx_calls_DoNotUseDirectly());
 #endif
 
+  bool writeProtectCode = true;
+  if (XRE_IsContentProcess()) {
+    writeProtectCode = StaticPrefs::javascript_options_content_process_write_protect_code();
+  }
+  JS_SetGlobalJitCompilerOption(cx, JSJITCOMPILER_WRITE_PROTECT_CODE, writeProtectCode);
+
   JS_SetGlobalJitCompilerOption(
       cx, JSJITCOMPILER_WATCHTOWER_MEGAMORPHIC,
       StaticPrefs::